metadata-io/src/main/java/com/linkedin/metadata/search/utils/ESUtils.java

```diff
@@ -107,12 +130,42 @@ public static BoolQueryBuilder buildConjunctiveFilterQuery(@Nonnull ConjunctiveC
    */
   @Nonnull
   public static QueryBuilder getQueryBuilderFromCriterion(@Nonnull Criterion criterion) {
+    String fieldName = toFacetField(criterion.getField());
+
+    Optional<String[]> pairMatch = Arrays.stream(EDITABLE_FIELD_TO_QUERY_PAIRS)
+        .filter(pair -> Arrays.stream(pair).anyMatch(pairValue -> pairValue.equals(fieldName)))
+        .findFirst();
+
+    if (pairMatch.isPresent()) {
+      final BoolQueryBuilder orQueryBuilder = new BoolQueryBuilder();
+      String[] pairMatchValue = pairMatch.get();
+      for(String field: pairMatchValue) {
+        Criterion criterionToQuery = new Criterion();
+        criterionToQuery.setCondition(criterion.getCondition());
+        criterionToQuery.setNegated(criterion.isNegated());
+        criterionToQuery.setValue(criterion.getValue());
+        criterionToQuery.setField(field + KEYWORD_SUFFIX);
+        orQueryBuilder.should(getQueryBuilderFromCriterionForSingleField(criterionToQuery));
+      }
+      return orQueryBuilder;
+    }
+
+    return getQueryBuilderFromCriterionForSingleField(criterion);
+  }
+  @Nonnull
+  public static QueryBuilder getQueryBuilderFromCriterionForSingleField(@Nonnull Criterion criterion) {
     final Condition condition = criterion.getCondition();
+    String fieldName = toFacetField(criterion.getField());
+
     if (condition == Condition.EQUAL) {
       // If values is set, use terms query to match one of the values
       if (!criterion.getValues().isEmpty()) {
+        if (BOOLEAN_FIELDS.contains(fieldName) && criterion.getValues().size() == 1) {
```

Comment:
if this is a boolean field then values shouldn't be anything other than 1, right? keeping the check regardless isn't a bad idea, just wanted to check for myself.

Reply:
yes correct!